app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/EnterMovementReferenceNumberController.scala

```diff
@@ -54,514 +49,118 @@ class EnterMovementReferenceNumberController @Inject() (
-                        evaluateMrnJourneyFlow(fillingOutClaim.signedInUserDetails, maybeDisplayDeclaration)
-                      )
-                      .leftMap(_ => Error("could not evaluate MRN flow"))
-                  displayDeclaration      <-
-                    EitherT.fromOption[Future](maybeDisplayDeclaration, Error("could not unbox display declaration"))
-                  _                       <- EitherT.liftF(
-                                               updateSession(sessionStore, request)(
-                                                 _.copy(journeyStatus =
-                                                   Some(
-                                                     fillingOutClaim.copy(draftClaim =
-                                                       newDraftClaim.copy(
-                                                         displayDeclaration = Some(displayDeclaration)
-                                                       )
-                                                     )
-                                                   )
-                                                 )
-                                               )
-                                             )
-                } yield mrnJourneyFlow
+  def enterMrnSubmit(): Action[AnyContent]  = mrnSubmit(false)
+  def changeMrnSubmit(): Action[AnyContent] = mrnSubmit(true)
 
-                result.fold(
-                  e => {
-                    logger.warn("could not get declaration information", e)
-                    Redirect(baseRoutes.IneligibleController.ineligible())
-                  },
-                  {
-                    case Left(_)  => Redirect(routes.CheckDeclarationDetailsController.checkDetails())
-                    case Right(_) => Redirect(routes.EnterImporterEoriNumberController.enterImporterEoriNumber())
-                  }
-                )
-            }
-          )
-      }
-    }
-
-  def changeMrnSubmit(): Action[AnyContent] =
+  def mrnSubmit(isCYA: Boolean): Action[AnyContent] =
     authenticatedActionWithSessionData.async { implicit request =>
-      withMovementReferenceNumberAnswer { (_, fillingOutClaim, _) =>
+      withAnswers[MovementReferenceNumber] { (fillingOutClaim, previousAnswer) =>
         EnterMovementReferenceNumberController
           .movementReferenceNumberForm(featureSwitch)
           .bindFromRequest()
           .fold(
-            requestFormWithErrors =>
+            formWithErrors =>
               BadRequest(
                 enterMovementReferenceNumberPage(
-                  requestFormWithErrors.copy(errors =
-                    Seq(EnterMovementReferenceNumberController.processFormErrors(requestFormWithErrors.errors))
-                  ),
-                  isAmend = true
+                  formWithErrors
+                    .copy(errors = Seq(EnterMovementReferenceNumberController.processFormErrors(formWithErrors.errors)))
```

Comment:
need to pass isAmend flag here, i think.

Reply:
Done